### PR TITLE
[FIX JENKINS-43848] -  Lack of cache-invalidation headers results in stale item list

### PR DESCRIPTION
**Pull request by josiahhaswell**
See [JENKINS-43848](https://issues.jenkins-ci.org/browse/JENKINS-43848).

### Proposed changelog entries

* Fix for JENKINS-43848: Browser caching response results in stale list
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention
@reviewbybees

### CODE REVIEW

**Comment by josiahhaswell** [comment]
There's some sort of whack white-space issue.  Those are not showing up in my local diff.  My Git settings are checkin with Unix line-feeds, checkout with local OS.

**Comment by oleg-nenashev** [comment]
@josiahhaswell Which IDE you use just in case?

**Comment by daniel-beck** [comment]
@josiahhaswell It's not line endings, your IDE reformats empty lines.

**Review by oleg-nenashev — DISMISSED** [review]
Although whitespace fixes are correct, they should not be fixed in the bugfix  since they complicate backporting. Likely you use IDEA and hence need to configure it to be less smart :(

**Comment by kzantow** [comment]
Also side note: if you use format: [FIX JENKINS-######] in the title, the issue tracker will automatically update things for you nicely

**Comment by oleg-nenashev** [comment]
@kzantow Yes, you are right. But we can always fix it by squash :)

**Comment by josiahhaswell** [comment]
Ok.  I had a version of IntelliJ with a bug wherein it would remove trailing whitespace no matter what was selected.  Upgraded and fixed.

File: test/src/test/java/hudson/model/ViewTest.java [hunk 1]
```diff
@@ -85,6 +90,26 @@
...
+        final Map<String, String> values = new HashMap<>();
+
+        for(NameValuePair p : responseHeaders) {
+            values.put(p.getName(), p.getValue());
+        }
+
+        String resp = values.get("Cache-Control");
+        assertThat(resp, is("no cache, no store, must revalidate"));
+        assertThat(values.get("Expires"), is("0"));
+        assertThat(values.get("Pragma"), is("no-cache"));
```
Comment (jglick) [inline test/src/test/java/hudson/model/ViewTest.java]: FWIW easier to write:

```java
WebResponse rsp = j.createWebClient().goTo("view/all/itemCategories", "application/json").getWebResponse();
assertEquals("no cache, no store, must revalidate", rsp.getResponseHeaderValue("Cache-Control"));
assertEquals("0", rsp.getResponseHeaderValue("Expires"));
assertEquals("no cache", rsp.getResponseHeaderValue("Pragma"));
```

**Review by jglick — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
🐝 and @reviewbybees done

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Opps, test needs to be fixed, hence 🐛 

```
Error
Expected: is "no cache, no store, must revalidate"
     but: was "no-cache, no-store, must-revalidate"
Stacktrace
java.lang.AssertionError: 
Expected: is "no cache, no store, must revalidate"
     but: was "no-cache, no-store, must-revalidate"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at hudson.model.ViewTest.testNoCacheHeadersAreSet(ViewTest.java:108)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.jvnet.hudson.test.JenkinsRule$2.evaluate(JenkinsRule.java:547)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Standard Output
=== Starting testNoCacheHeadersAreSet(hudson.model.ViewTest)
Standard Error
Aug 08, 2017 7:59:09 PM org.jvnet.hudson.test.JenkinsRule createWebServer
INFO: Running on http://localhost:41088/jenkins/
Aug 08, 2017 7:59:09 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started initialization
Aug 08, 2017 7:59:09 PM jenkins.InitReactorRunner$1 onAttained
INFO: Listed all plugins
Aug 08, 2017 7:59:09 PM jenkins.InitReactorRunner$1 onAttained
INFO: Prepared all plugins
Aug 08, 2017 7:59:09 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started all plugins
Aug 08, 2017 7:59:09 PM jenkins.InitReactorRunner$1 onAttained
INFO: Augmented all extensions
Aug 08, 2017 7:59:10 PM jenkins.InitReactorRunner$1 onAttained
INFO: Loaded all jobs
Aug 08, 2017 7:59:10 PM jenkins.InitReactorRunner$1 onAttained
INFO: Completed initialization
Aug 08, 2017 7:59:10 PM jenkins.model.Jenkins cleanUp
INFO: Stopping Jenkins
Aug 08, 2017 7:59:11 PM jenkins.model.Jenkins cleanUp
INFO: Jenkins stopped
```

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**